massa-execution-worker/src/tests/scenarios_mandatories.rs

```diff
@@ -1,173 +1,193 @@
-                SCELedgerEntry {
-                    balance: Amount::from_str("878").unwrap(),
-                    opt_module: Some("bytecodebytecode".into()),
-                    data: vec![
-                        (
-                            massa_hash::hash::Hash::compute_from("key_testD".as_bytes()),
-                            "test4_data".into(),
-                        ),
-                        (
-                            massa_hash::hash::Hash::compute_from("key_testE".as_bytes()),
-                            "test5_data".into(),
-                        ),
-                    ]
-                    .into_iter()
-                    .collect(),
-                },
-            ),
-        ]
-        .into_iter()
-        .collect(),
-    )
+fn get_sample_ledger() -> Result<(Arc<RwLock<FinalLedger>>, NamedTempFile), LedgerError> {
+    let mut initial: BTreeMap<Address, Amount> = Default::default();
+    initial.insert(get_random_address(), Amount::from_str("129").unwrap());
+    initial.insert(get_random_address(), Amount::from_str("878").unwrap());
+    let (cfg, tempfile) = LedgerConfig::sample(&initial);
+    Ok((Arc::new(RwLock::new(FinalLedger::new(cfg)?)), tempfile))
 }
 
-#[tokio::test]
+#[test]
 #[serial]
-async fn test_execution_basic() {
-    let (_config_file_keepalive, settings) = get_sample_settings();
-    assert!(start_controller(settings, None).await.is_ok());
+fn test_execution_basic() {
+    let (sample_ledger, _keep) = get_sample_ledger().unwrap();
+    let (_, _) = start_execution_worker(ExecutionConfig::default(), sample_ledger);
 }
 
-#[tokio::test]
+#[test]
 #[serial]
-async fn test_execution_shutdown() {
-    let (_config_file_keepalive, settings) = get_sample_settings();
-    let (_command_sender, _event_receiver, manager) = start_controller(settings, None)
-        .await
-        .expect("Failed to start execution.");
-    manager.stop().await.expect("Failed to stop execution.");
+fn test_execution_shutdown() {
+    let (sample_ledger, _keep) = get_sample_ledger().unwrap();
+    let (mut manager, _) = start_execution_worker(ExecutionConfig::default(), sample_ledger);
+    manager.stop()
 }
 
-#[tokio::test]
+#[test]
 #[serial]
-async fn test_sending_command() {
-    let (_config_file_keepalive, settings) = get_sample_settings();
-    let (command_sender, _event_receiver, manager) = start_controller(settings, None)
-        .await
-        .expect("Failed to start execution.");
-    command_sender
-        .update_blockclique(Default::default(), Default::default())
-        .await
-        .expect("Failed to send command");
-    manager.stop().await.expect("Failed to stop execution.");
+fn test_sending_command() {
+    let (sample_ledger, _keep) = get_sample_ledger().unwrap();
+    let (mut manager, controller) =
+        start_execution_worker(ExecutionConfig::default(), sample_ledger);
+    controller.update_blockclique_status(Default::default(), Default::default());
```

Comment:
is there a way to check that it has indeed been updated ?

Reply:
the test `generate_events` does that, that one is used to check if it doesn't crash with an empty clique